control-plane/cni/main.go

```diff
@@ -22,36 +26,58 @@ import (
+
+	// annotationInject is the key of the annotation that controls whether
+	// injection is explicitly enabled or disabled for a pod.
+	annotationInject = "consul.hashicorp.com/connect-inject"
+	// annotationCNIProxyConfig stores iptables.Config information so that the CNI plugin can use it to apply iptables rules.
+	annotationCNIProxyConfig = "consul.hashicorp.com/cni-proxy-config"
+	// retries is the number of backoff retries to attempt while waiting for an cni-proxy-config annotation to poplulate.
+	retries = 10
+	// dnsServiceHostEnvSuffix is the suffix that is used to get the DNS host IP. The DNS IP is saved as an environment variable
+	// with prefix + suffix. The prefix is passed in as cli flag to the endpoints controller which is then passed on in the
+	// cni-proxy-config annotation so that the CNI plugin can use it. The DNS environment variable usually looks like:
+	// CONSUL_CONSUL_DNS_SERVICE_HOST but the prefix can change depending on the helm install.
+	dnsServiceHostEnvSuffix = "DNS_SERVICE_HOST"
 )
 
 type CNIArgs struct {
+	// types.CommonArgs are args that are passed as part of the CNI standard.
 	types.CommonArgs
-	IP                         net.IP
-	K8S_POD_NAME               types.UnmarshallableString
-	K8S_POD_NAMESPACE          types.UnmarshallableString
+	// IP address assigned to the pod from a previous plugin.
+	IP net.IP
+	// K8S_POD_NAME is the pod that the plugin is running for.
+	K8S_POD_NAME types.UnmarshallableString
+	// K8S_POD_NAMESPACE is the namespace that the plugin is running for.
+	K8S_POD_NAMESPACE types.UnmarshallableString
+	// K8S_POD_INFRA_CONTAINER_ID is the runtime container ID that the pod runs under.
 	K8S_POD_INFRA_CONTAINER_ID types.UnmarshallableString
 }
 
 // PluginConf is is the configuration used by the plugin.
 type PluginConf struct {
 	// NetConf is the CNI Specification configuration for standard fields like Name, Type,
-	// CNIVersion and PrevResult
+	// CNIVersion and PrevResult.
 	types.NetConf
 
+	// RuntimeConfig is the config passed from the kubelet to plugin at runtime.
 	RuntimeConfig *struct {
 		SampleConfig map[string]interface{} `json:"sample_config"`
 	} `json:"runtime_config"`
 
+	// Name of the plugin (consul-cni).
 	Name string `json:"name"`
-	// Type of plugin (consul-cni)
+	// Type of plugin (consul-cni).
 	Type string `json:"type"`
-	// CNIBinDir is the location of the cni config files on the node. Can bet as a cli flag.
+	// CNIBinDir is the location of the cni config files on the node. Can be set as a cli flag.
 	CNIBinDir string `json:"cni_bin_dir"`
 	// CNINetDir is the locaion of the cni plugin on the node. Can be set as a cli flag.
 	CNINetDir string `json:"cni_net_dir"`
+	// DNSPrefix is used to determine the Consul Server DNS IP. The IP is set as an environment variable and the prefix allows us
+	// to search for it. The DNS IP is determined using the prefix and the dnsServiceHostEnvSuffix constant.
+	DNSPrefix string `json:"dns_prefix"`
```

Comment:
Why do we have two config structs with a lot of duplicated information? Do you think we could somehow consolidate them?

Reply:
Will look at this at a later date, maybe when multus is done.